### PR TITLE
Add iconOnly modifier to Button

### PR DESCRIPTION
**Pull request by matzimowski**
## changes:
- `iconOnly` modifier introduced
- `hover` and `toggle hover` state of `Button` removed from docs
- `RoundedButton` component became deprecated

we want to make it work in both ways: in React and in plain html

Screenshot 2020-05-14 at 13 18 31
Screenshot 2020-05-14 at 13 18 39
Screenshot 2020-05-14 at 13 18 43

### CODE REVIEW

File: src/components/buttons/Button.spec.jsx [hunk 1]
```diff
@@ -54,6 +54,25 @@ test('icon', () => {
...
+  const icon = <span>:P</span>;
+  const button = shallow(
+    <Button icon={icon} iconOnly>
+      Some text
+    </Button>
+  );
+
+  expect(button.contains(icon)).toEqual(true);
+  expect(button.find('.sg-button--icon-only')).toHaveLength(1);
+  expect(button.find('.sg-button__icon')).toHaveLength(1);
```
Comment (Malinicz) [inline src/components/buttons/Button.spec.jsx:67]: maybe add additional expect to check that the text is not rendered?

File: src/components/buttons/Button.spec.jsx [hunk 2]
```diff
@@ -54,6 +54,25 @@ test('icon', () => {
...
+});
+
+test('icon only when no icon', () => {
+  const button = shallow(<Button iconOnly>Some text</Button>);
```
Comment (Malinicz) [inline src/components/buttons/Button.spec.jsx:71]: is it possible to type it in a way that when you pass `iconOnly`, you must pass also `icon`?

Comment (bartoszlorek) [reply src/components/buttons/Button.spec.jsx:71]: it's already typed but tests are flow-free

Comment (Malinicz) [reply src/components/buttons/Button.spec.jsx:71]: yepp, but I think we don't need such test then? idk

Comment (matzimowski) [reply src/components/buttons/Button.spec.jsx]: good point, removed

File: src/components/buttons/Button.spec.jsx (hunk 1 above)
Comment (matzimowski) [reply src/components/buttons/Button.spec.jsx:67]: but children are optional already, if you pass some text i will be rendered but not visible

**Pull request merged**